api/server/server.go

```diff
@@ -1137,8 +1137,17 @@ func (s *Server) getContainersByName(version version.Version, w http.ResponseWri
 		return fmt.Errorf("Missing parameter")
 	}
 
+	if err := parseForm(r); err != nil {
+		return err
+	}
+
+	no_err := r.Form.Get("no_err")
 	containerJSON, err := s.daemon.ContainerInspect(vars["name"])
 	if err != nil {
+		if strings.Contains(strings.ToLower(err.Error()), "no such id") && no_err == "1" {
+			w.WriteHeader(http.StatusNoContent)
```

Comment:
If we can't generate a 404 w/o logging an 'error' then keep it as an 'error', but 404 is what should be returned from the daemon.  I don't understand your comment about the CLI showing the error.  Yes it'll show an error but only after both 'inspect' calls are done and both fail.  Which is correct.